exp/etcdrestore/controllers/etcdmachinesnapshot_controller.go

```diff
@@ -60,18 +67,224 @@ func (r *EtcdMachineSnapshotReconciler) SetupWithManager(_ context.Context, mgr
 //+kubebuilder:rbac:groups=turtles-capi.cattle.io,resources=etcdmachinesnapshots/finalizers,verbs=update
 
 // Reconcile reconciles the EtcdMachineSnapshot object.
-func (r *EtcdMachineSnapshotReconciler) Reconcile(_ context.Context, _ ctrl.Request) (ctrl.Result, error) {
-	return ctrl.Result{}, nil
+func (r *EtcdMachineSnapshotReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+
+	etcdMachineSnapshot := &snapshotrestorev1.EtcdMachineSnapshot{ObjectMeta: metav1.ObjectMeta{
+		Name:      req.Name,
```

Comment:
no need to setup the name and namespace here, it will be filled out later by Get() request.